manager/orchestrator/constraintenforcer/constraint_enforcer.go

```diff
@@ -83,10 +84,15 @@ func (ce *ConstraintEnforcer) rejectNoncompliantTasks(node *api.Node) {
 		log.L.WithError(err).Errorf("failed to list tasks for node ID %s", node.ID)
 	}
 
-	var availableMemoryBytes, availableNanoCPUs int64
+	available := &api.Resources{}
+	fakeStore := make([]*api.GenericResource, 0)
+
 	if node.Description != nil && node.Description.Resources != nil {
-		availableMemoryBytes = node.Description.Resources.MemoryBytes
-		availableNanoCPUs = node.Description.Resources.NanoCPUs
+		available = node.Description.Resources.Copy()
+	}
+
+	if available.Generic == nil {
+		available.Generic = make([]*api.GenericResource, 0)
```

Comment:
Yes, for all practical purposes a `nil` slice is the same as an empty one in Go. You can `range` over it, `append` to it, and so on.